api/src/utils/validate-diff.ts

```diff
@@ -7,11 +7,15 @@ const deepDiffSchema = Joi.object({
 		.valid(...Object.values(DiffKind))
 		.required(),
 	path: Joi.array().items(Joi.string()),
```

Comment:
```suggestion
	path: Joi.array().items(Joi.alternatives().try(Joi.string(), Joi.number())),
```

The path may contain the integer index when an array value is edited, which currently errors out with `"diff.fields[0].diff[1].path[2]" must be a string`.

### Partial Diff
```js
fields: [
  {
    collection: "test",
    field: "name",
    diff: [
      {
        kind: "A",
        path: ["meta", "translations"],
        index: 2,
        item: {
          kind: "D",
          lhs: { language: "nl-NL", translation: "naam" },
        },
      },
      {
        kind: "E",
        path: ["meta", "translations", 1, "language"],
        lhs: "es-ES",
        rhs: "nl-NL",
      },
      {
        kind: "E",
        path: ["meta", "translations", 1, "translation"],
        lhs: "nombre",
        rhs: "naam",
      },
    ],
  },
];
```

Reply:
Wow, good catch! Thanks ❤️